### PR TITLE
feature(Analysis of the movements of the auxiliary Cashboxes)

### LINKED ISSUE #3871: Analysis of the movements of the auxiliary cashboxes
This issue proposes to add a new report, to allow the monthly follow-up of the movements of the auxiliary cashboxes

### PR DESCRIPTION
**Pull request by lomamech**
- improve component bh-cashbox-select and the filter isAuxiliary
- Implement API and Client side of Report

closes #3871

### CODE REVIEW

**Comment by kwilu** [comment]
Hello @lomamech, it seems like you have referenced #3871 in your pull request description, but you have not referenced them in your commit message description(s). When you reference an issue in a commit message, it automatically closes the corresponding issue when the commit is merged.

Please run `git commit --amend` in your command line client to amend your commit message description with `Fixes #3871.`.

An example of a correctly-formatted commit:
```
commit fabd5e450374c8dde65ec35f02140383940fe146
Author: kwilu
Date:   Sat Mar 18 13:42:40 2017 -0700

    pull requests: Check PR commits reference when issue is referenced.

    Fixes #51.
```

Thank you for your contributions to BHIMA!

**Comment by lomamech** [comment]
@jniles  all checks have passed

**Comment by lomamech** [comment]
image

File: client/src/modules/reports/generate/analysisAuxiliaryCash/analysisAuxiliaryCash.config.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+angular.module('bhima.controllers')
+  .controller('analysisAuxiliaryCashController', AnalysisAuxiliaryCashController);
+
+AnalysisAuxiliaryCashController.$inject = [
+  '$sce', 'NotifyService', 'BaseReportService', 'AppCache', 'reportData', '$state', 'AccountService',
+];
+
+function AnalysisAuxiliaryCashController($sce, Notify, SavedReports, AppCache, reportData, $state, Accounts) {
+  const vm = this;
+  const cache = new AppCache('monthlyBalance');
```
Comment (mbayopanda) [inline client/src/modules/reports/generate/analysisAuxiliaryCash/analysisAuxiliaryCash.config.js]: Does the cache is for `montlyBalance` ?

**Review by mbayopanda — COMMENTED** [review]
👍 Good job @lomamech  

Could fix the conflict so that we will merge the branch

**Review by mbayopanda — APPROVED** [review]
LGTM 👍 

bors r+

**Comment by jniles** [comment]
@mbayopanda it looks like Travis/Protractor is messing with us again.  Do you want to work on this or would you like me to?

**Comment by mbayopanda** [comment]
I am trying to fix it

**Comment by mbayopanda** [comment]
@lomamech could you update this branch, so then the build will succeed

**Comment by jniles** [comment]
@mbayopanda, I think that bors might just work, since bors will run on the merged branches.  Let's try it 👍

**Pull request merged**